compiler/rustc_hir_typeck/src/upvar.rs

```diff
@@ -712,10 +712,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     }
                 }
 
-                unreachable!(
-                    "we captured two identical projections: capture1 = {:?}, capture2 = {:?}",
-                    capture1, capture2
-                );
+                // return Equal for two identical projections
```

Comment:
it looks like this could be a `delay_span_bug` instead? :thinking:

Reply:
yes, I will make a change.

Reply:
I add `delay_span_bug` and also keep the `std::cmp::Ordering::Equal` so that we keep minimal change since we need to return `Ordering` type for the `sort_by` function.